message_ix_models/tests/model/material/test_build.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import pytest
+
+from message_ix_models.model.material import build
+from message_ix_models.model.structure import get_codes
+from message_ix_models.testing import bare_res
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "regions_arg, regions_exp",
+    [
+        # ("R11", "R11"),
+        ("R12", "R12"),
+    ],
+)
+@pytest.mark.parametrize("material", [None])
+def test_make_spec(regions_arg, regions_exp, material):
+    # The spec can be generated
+    spec = build.make_spec(regions_arg, material)
+
+    # The required elements of the "node" set match the configuration
+    nodes = get_codes(f"node/{regions_exp}")
+    expected = list(map(str, nodes[nodes.index("World")].child))
+    assert expected == spec["require"].set["node"]
+
+
+@pytest.mark.parametrize(
+    "regions, years, relations, solve",
+    [
+        ("R12", "B", "B", False),
+    ],
```

Comment:
```suggestion
    [
        ("R12", "B", "B", False),
        ("R12", "B", "B", True),
    ],
```
A parametrization like this is great, but would likely be even more beneficial if the function parameters actually received different values. Also, this would add line 50 of this file to the tests :)

Reply:
At the moment, the bare RES is not sufficiently parametrized to allow for sensible model solve tests on a MESSAGEix-Materials build. MESSAGEix-Materials depends on many elements of MESSAGEix-GLOBIOM. We would need to add several supply dummies for all the fuels that MESSAGEix-Materials needs for example. We also rely on the GLOBIOM emulator lookup table. So I am expecting quite a workload to emulate all of that in the bare RES. Maybe a XFAIL test would be possible already, but I have not tried that yet.